### PR TITLE
Merge coverages

### PR DESCRIPTION
**Pull request by mehmetalisavas**
Coverage results doesn't show total coverage of all tests, we need to merge as atomic into 1 coverage file.

### CODE REVIEW

**Comment by mehmetalisavas** [comment]
@rjeczalik i've take a look for your merge coverage package but didn't find main in your package. So  for now we are using 'wadey/gocovmerge' package to merge coverage files.

**Comment by rjeczalik** [comment]
@mehmetalisavas 👍 

mcov is a fork of wadey/gocovmerge, for use within the tests - if one test creates > 1 coverage profile, mcov merges them back to 1 upon exit.

**Comment by rjeczalik** [comment]
Basically it's two functions, like in [this example](https://github.com/rjeczalik/mcov/tree/master/testdata/src/example):

- [one to hook into external process to inject coverage](https://github.com/rjeczalik/mcov/blob/master/testdata/src/example/helper_test.go#L67)
- [second one to merge profiles upon exit](https://github.com/rjeczalik/mcov/blob/master/testdata/src/example/helper_test.go#L56)

This was suitable for kd, since now processes start by klientctl/main_test.go does not produce coverage profiles, thus there's nothing to merge. More details [are here](https://github.com/koding/koding/pull/10238#issue-199439872).

**Review by cihangir — APPROVED** [review]

**Review by ppknap — APPROVED** [review]

File: scripts/gotests.sh
Comment (rjeczalik) [inline scripts/gotests.sh]: nitpick: can be

```go
function merge () {
     local folder=$(createFolder $1)
...
```

so the function won't overwrite global variable

File: scripts/gotests.sh
Comment (rjeczalik) [inline scripts/gotests.sh]: this line is not needed

File: scripts/gotests.sh
Comment (rjeczalik) [inline scripts/gotests.sh]: we can just echo this directly

**Review by rjeczalik — APPROVED** [review]

**Comment by cihangir** [comment]
This Pr has everyone's approval, @mehmetalisavas please send the fixes in another pr as soon as possible, merging this.

**Pull request merged**
